lib/cache/access_list_test.go

```diff
@@ -274,6 +276,111 @@ func TestAccessListReviews(t *testing.T) {
+	names := []string{"apple-list", "banana-access", "cherry-management", "apple-admin", "zebra-test"}
+
+	for i, name := range names {
+		al := newAccessList(t, name, clock)
+		// Set different audit dates to test sorting
+		auditDate := clock.Now().Add(time.Duration(i) * (time.Hour) * 24)
+		// add arbitrary date so we can make sure its not just sorted by name
+		if name == "banana-access" {
+			auditDate = clock.Now().Add(100 * (time.Hour) * 24)
+		}
+		al.Spec.Audit.NextAuditDate = auditDate
+
+		_, err := p.accessLists.UpsertAccessList(ctx, al)
+		require.NoError(t, err)
+	}
+
+	testCases := []struct {
+		name            string
+		search          string
+		sortBy          *types.SortBy
+		expectedNames   []string
+		pageSize        int
+		expectedNextKey string
+		startKey        string
+	}{
+		{
+			name:          "no filter - all lists",
+			expectedNames: []string{"apple-admin", "apple-list", "banana-access", "cherry-management", "zebra-test"},
+		},
+		{
+			name:          "sort by name reverse",
+			sortBy:        &types.SortBy{Field: "name", IsDesc: true},
+			expectedNames: []string{"zebra-test", "cherry-management", "banana-access", "apple-list", "apple-admin"},
+		},
+		{
+			name:          "sort by audit date",
+			sortBy:        &types.SortBy{Field: "auditNextDate", IsDesc: false},
+			expectedNames: []string{"apple-list", "cherry-management", "apple-admin", "zebra-test", "banana-access"},
+		},
+		{
+			name:          "sort by audit date reverse",
+			sortBy:        &types.SortBy{Field: "auditNextDate", IsDesc: true},
+			expectedNames: []string{"banana-access", "zebra-test", "apple-admin", "cherry-management", "apple-list"},
+		},
+		{
+			name:            "paginated results",
+			expectedNames:   []string{"apple-admin", "apple-list"},
+			pageSize:        2,
+			expectedNextKey: "banana-access",
+		},
+		{
+			name:            "paginated results reverse",
+			expectedNames:   []string{"zebra-test", "cherry-management", "banana-access"},
+			sortBy:          &types.SortBy{Field: "name", IsDesc: true},
+			pageSize:        3,
+			expectedNextKey: "apple-list",
+		},
+		{
+			name:          "with search",
+			search:        "apple",
+			expectedNames: []string{"apple-admin", "apple-list"},
+		},
+		{
+			name:          "with startKey",
+			startKey:      "banana-access",
+			expectedNames: []string{"banana-access", "cherry-management", "zebra-test"},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			require.EventuallyWithT(t, func(collect *assert.CollectT) {
+				results, nextToken, err := p.cache.ListAccessListsWithFilter(ctx, &accesslistv1.ListAccessListsWithFilterRequest{
+					PageSize:  int32(tc.pageSize),
+					PageToken: tc.startKey,
+					Filter: &accesslistv1.ListAccessListsFilter{
+						Search: tc.search,
+					},
+					SortBy: tc.sortBy,
+				})
+				assert.NoError(collect, err)
+				assert.Equal(t, tc.expectedNextKey, nextToken)
```

Comment:
Careful mixing `collect` and `t`.

I recommend you rename `collect` to `t` so that it's impossible to use the wrong one.